pkg/scheduler/scheduler.go

```diff
@@ -32,19 +32,21 @@ type Scheduler interface {
 }
 
 type RemoteScheduler struct {
+	inventory      cluster.Inventory
 	inventoryWatch InventoryWatcher
 	workerFactory  WorkerFactory
 	mothershipCfg  reconciler.MothershipReconcilerConfig
 	poolSize       int
 	logger         *zap.SugaredLogger
 }
 
-func NewRemoteScheduler(inventoryWatch InventoryWatcher, workerFactory WorkerFactory, mothershipCfg reconciler.MothershipReconcilerConfig, workers int, debug bool) (Scheduler, error) {
+func NewRemoteScheduler(inventory cluster.Inventory, inventoryWatch InventoryWatcher, workerFactory WorkerFactory, mothershipCfg reconciler.MothershipReconcilerConfig, workers int, debug bool) (Scheduler, error) {
 	l, err := logger.NewLogger(false)
```

Comment:
We could add a `NewWatcher()` method to the inventory interface and move the creation o the watcher into the inventory implementation. This would help to simplify this method signature and also bundle the creation of the `InventoryWatcher` with the underneath used inventory implementation (e.g. database-lookup based, or in case of an in-memory inventory it could be callback based) .. what do you think?

Reply:
Other option could be to adda a getter for the inventory to the `InventoryWatcher`, e.g.

     func (w *DefaultInventoryWatcher) Inventory() *cluster.Inventory {
          return w.inventory
     }